apps/server/lib/lexical/server/code_intelligence/completion.ex

```diff
@@ -58,7 +58,11 @@ defmodule Lexical.Server.CodeIntelligence.Completion do
         do_end_snippet = "do\n  $0\nend"
 
         env
-        |> Builder.snippet(do_end_snippet, label: "do/end block")
+        |> Builder.snippet(
+          do_end_snippet,
+          label: "do/end block",
+          filter_text: "do"
```

Comment:
shouldn't the filter_text be whatever the prefix was? (likely `d`) ?

Reply:
AFAICT there's no hard and fast rules to filter texts, they just have to fuzzy match well to the expected inputs.

I ran a quick test on this with vscode's client:
* `filter_text: "do"` provides the completion for both `d|` and `do|`.
* `filter_text: "d"` only provides the completion for `d|` (doesn't pass the fuzzy match margin).

Hence I'd say stick with `filter_text: "do"`.

I'll have to get myself setup with neovim so I can validate there as well.

Broader scope, perhaps an LSP client test suite's in order at some point - that's a big uplift though.

Reply:
vscode has been the most finicky with this, emacs already works.